src/handlers.ts

```diff
@@ -0,0 +1,33 @@
+import {DispatchError} from "@polkadot/types/interfaces";
+
+export class ProsopoEnvError extends Error {
+  constructor(error: Error | string, context?: string, ...params: any[]) {
+    const isError = error instanceof Error;
+    super(isError ? error.message : error);
+    this.name = context && `${ProsopoEnvError.name}@${context}` || ProsopoEnvError.name;
+    if (isError) {
+      this.cause = error;
+    }
+    // TODO: if env.debug
+    console.error('\n********************* ERROR *********************\n');
+    console.error(this.cause, this.stack, ...params);
+  }
+}
+
+export class ProsopoContractError extends Error {
+  constructor(error: DispatchError | string, context?: string, ...params: any[]) {
+    if (typeof error === "string") {
```

Comment:
if (error instanceof DispatchError) as above (would not care about instance of String ambiguity

Reply:
`instanceof DispatchError` doesn't work because it's `DispatchError` is a type and doesn't exist after compilation. 

![image](https://user-images.githubusercontent.com/7630866/174258777-3e1b95f0-4ed1-45b2-82a8-9226119e677f.png)

Getting the keys of `DispatchError` and checking them all [was not going to be easy](https://stackoverflow.com/questions/43909566/get-keys-of-a-typescript-interface-as-array-of-strings), which is why I checked for `string`.